cmake/findBLIS.cmake

```diff
@@ -21,26 +21,35 @@
 #
 # ########################################################################
 
-file(GLOB BLIS_AOCL_LIBS /opt/AMD/aocl/aocl-linux-aocc-*/aocc/lib_ILP64/libblis-mt.a)  # Possible location 1
-file(GLOB BLIS_LOCAL_LIB /usr/local/lib/libblis.a)  # Possible location 2
+file(GLOB BLIS_AOCL_LIB_1 /opt/AMD/aocl/aocl-linux-gcc-*/gcc/lib_ILP64/libblis-mt.a)  # Possible location 1
+file(GLOB BLIS_AOCL_LIB_2 /opt/AMD/aocl/aocl-linux-aocc-*/aocc/lib_ILP64/libblis-mt.a)  # Possible location 2
+file(GLOB BLIS_LOCAL_LIB /usr/local/lib/libblis.a)  # Possible location 3
 file(GLOB BLIS_DEPS_LIB ${CMAKE_SOURCE_DIR}/build/deps/blis/lib/libblis.a)  # wget location
-set(BLIS_LIBS ${BLIS_LOCAL_LIB} ${BLIS_AOCL_LIBS} ${BLIS_DEPS_LIB})
-list(REVERSE BLIS_LIBS)
-list (GET BLIS_LIBS 0 BLIS_LIB)
+
+set(BLIS_LIBS ${BLIS_AOCL_LIB_1} ${BLIS_AOCL_LIB_2} ${BLIS_DEPS_LIB} ${BLIS_LOCAL_LIB})
+
+# Remove empty strings
+foreach(item IN LISTS BLIS_LIBS)
+    if(NOT item STREQUAL "")
+        list(APPEND BLIS_LIBS_NON_EMPTY_LIST "${item}")
+    endif()
+endforeach()
```

Comment:
I don't understand the value of this pattern over a simple if /else if with precedence in the order of checking.  Am I missing something here.

Reply:
I did not want to change their pattern of finding the path to the AOCL BLIS, the only changes I did is adding the path to find AOCL BLIS 4.2 and removed the reverse and instead added remove empty strings.

Reply:
You can change to simple if/else if you want. Although it's simple but it makes code longer.

Reply:
Changed to simple if/else. Thanks!